homeassistant/components/sensor/nextbus.py

```diff
@@ -0,0 +1,271 @@
+    """
+
+    def __init__(self, agency, route, stop, name=None):
+        """Initialize sensor with all required config."""
+        self.agency = agency
+        self.route = route
+        self.stop = stop
+        self._custom_name = name
+        self._name = '{} - {}'.format(agency, route)
+
+        # Initialize a unique logger
+        self._logger = _LOGGER.getChild('.'.join((
+            self.__class__.__name__,
+            self.agency,
+            self.route,
+            self.stop,
+        )))
+
+        # set up default state attributes
+        self._state = 0
+        self._attributes = {}
+
+    @property
+    def name(self):
+        """Return sensor name.
+
+        Uses an auto generated name based on the data from the API unless a
+        custom name is provided in the configuration.
+        """
+        if self._custom_name:
+            return self._custom_name
+
+        return self._name
+
+    @property
+    def state(self):
+        """Return current state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return ATTR_MINUTES
+
+    @property
+    def device_state_attributes(self):
+        """Return additional state attributes."""
+        return self._attributes
+
+    @property
+    def icon(self):
+        """Return icon to be used for this sensor."""
+        # Would be nice if we could determine if the line is a train or bus
+        # however that doesn't seem to be available to us. Using bus for now.
+        return ICON
+
+    def update(self):
+        """Update sensor with new departures times."""
+        # Note: using Multi because there is a bug with the single stop impl
+        predictions = get_predictions(
+            [{
+                'stop_tag': int(self.stop),
+                'route_tag': self.route,
+            }],
+            self.agency,
+        )
+
+        self._logger.debug('Predictions results: %s', predictions)
+
+        if 'Error' in predictions:
+            self._logger.error('Could not get predictions: %s', predictions)
+
+        if not predictions.get('predictions'):
+            self._logger.info('No predictions available')
+            self._state = None
+            # Remove attributes that may now be outdated
+            self._attributes.pop('Upcoming', None)
+            return
+
+        predictions = predictions['predictions']
+
+        # Set detailed attributes
+        self._attributes.update({
+            'Agency': predictions.get('agencyTitle'),
+            'Route': predictions.get('routeTitle'),
+            'Direction': predictions.get('direction', {}).get('title'),
+            'Stop': predictions.get('stopTitle'),
+        })
+
+        # Sometimes message will return a dict, others a list of dicts
+        message = predictions.get('message')
+        if isinstance(message, dict):
+            self._attributes['Message'] = message.get('text', '')
```

Comment:
What does the message contain?

Reply:
Notices about delays and such. It's an inconsistent schema, which is rather annoying. Eg:

```
{
  "message": {"text": "Everything is on fire"}
}
```
Or
```
{
  "message": [
    {"text": "Everything is on fire"},
    {"text": "This is ok"}
  ]
}
```

I'm trying to display these to the user in a way that they can tell they are different messages. I could use `.` to join, but when a message is multiple sentences, it makes it unclear where a message ends and a new one begins.

Reply:
State attributes that aren't included in specific entity models as properties are mainly useful for automations. Please consider limiting them and structuring them here so that they are easy to use in templates and automations.

Reply:
I see. I'm not sure if there is a better way to display this additional information within Home Assistant. Any other ideas?

Reply:
What does the order in the list mean? Is latest to oldest or reversed? Maybe only show the latest message?

Reply:
They may be ordered, but it only contains "current" messages. For example, the other week there was a schedule change due to a parade along with a delay on the line that I was getting status for. They showed up as two messages in the list, but both were definitely relevant.